pkg/spdx/implementation.go

```diff
@@ -193,31 +193,35 @@ func (di *spdxDefaultImplementation) ReadArchiveManifest(manifestPath string) (m
 
 // getImageReferences gets a reference string and returns all image
 // references from it
-func getImageReferences(referenceString string) ([]ImageReferenceInfo, error) {
+func getImageReferences(referenceString string) (*ImageReferenceInfo, error) {
 	ref, err := name.ParseReference(referenceString)
 	if err != nil {
 		return nil, fmt.Errorf("parsing image reference %s: %w", referenceString, err)
 	}
 
-	images := []ImageReferenceInfo{}
+	images := &ImageReferenceInfo{
+		Images: []ImageReferenceInfo{},
+	}
 
 	descr, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
 	if err != nil {
 		return nil, fmt.Errorf("fetching remote descriptor: %w", err)
 	}
 
-	// If we got a digest, we reuse it as is
-	if _, ok := ref.(name.Digest); ok {
-		images = append(images, ImageReferenceInfo{Digest: ref.(name.Digest).String()})
-		logrus.Infof("Adding image %s", ref)
-		return images, nil
-	}
-
 	// If the reference is not an image, it has to work as a tag
-	tag, ok := ref.(name.Tag)
-	if !ok {
-		return nil, fmt.Errorf("could not cast tag from reference %s: %w", referenceString, err)
+	tag, tok := descr.Ref.(name.Tag)
+	dig, dok := descr.Ref.(name.Digest)
+	if !tok && !dok {
+		return nil, fmt.Errorf("could not cast tag or digest from reference %s: %w", referenceString, err)
 	}
+
+	// Add the main digest
+	images.Digest = fmt.Sprintf(
+		"%s/%s@%s:%s",
+		tag.RegistryStr(), tag.RepositoryStr(),
```

Comment:
If `tok == false` but `dok == true` then the above statement will not cover it and we probably read garbage from `tag.RegistryStr()`/`tag.RegistryStr()`.

Reply:
OK, I've split the index and single-image info extraction functions into two separate ones to simplify the logic. This logic in particular is now removed and we have much cleaner code paths, PTAL!